drivers/overlay/overlay.go

```diff
@@ -1486,3 +1595,80 @@ func nameWithSuffix(name string, number int) string {
 	}
 	return fmt.Sprintf("%s%d", name, number)
 }
+
+func (d *Driver) getAdditionalLayerPath(dgst digest.Digest, ref string) (string, error) {
+	refelem := base64.StdEncoding.EncodeToString([]byte(ref))
+	for _, ls := range d.options.layerStores {
+		var elems []string
+		if ls.withReference {
+			elems = append(elems, refelem)
+		}
+		elems = append(elems, dgst.String())
+		target := filepath.Join(append([]string{ls.path}, elems...)...)
```

Comment:
This code is not super readable (to me). What about something like
```go
ref := ""
if ls.withReference {
   ref = refElem
}
target := path.Join(ls.path, ref, dgst.String())
```
instead? (untested)